traffic_ops/testing/api/v4/cachegroups_test.go

```diff
@@ -455,22 +510,33 @@ func UpdateTestCacheGroups(t *testing.T) {
 		t.Fatal("got an empty response for cachegroups")
 	}
 	cg = resp[0]
-	if !reflect.DeepEqual(expectedFallbacks, *cg.Fallbacks) {
+	if cg.Fallbacks == nil {
+		t.Error("Cache Group returned by Traffic Ops had null or undefined fallbacks")
+	} else if !reflect.DeepEqual(expectedFallbacks, *cg.Fallbacks) {
 		t.Errorf("failed to update fallbacks (expected = %v, actual = %v)", expectedFallbacks, *cg.Fallbacks)
 	}
+	if cg.ID == nil {
+		t.Fatal("Cache Group returned by Traffic Ops had null or undefined ID")
+	}
 
 	const topologyEdgeCGName = "topology-edge-cg-01"
-	resp, _, err = TOSession.GetCacheGroupNullableByNameWithHdr(topologyEdgeCGName, nil)
+	resp, _, err = TOSession.GetCacheGroupByName(topologyEdgeCGName, nil)
 	if err != nil {
-		t.Errorf("cannot GET CacheGroup by name: '$%s', %v", topologyEdgeCGName, err)
+		t.Fatalf("cannot GET CacheGroup by name: '$%s', %v", topologyEdgeCGName, err)
 	}
 	if len(resp) == 0 {
 		t.Fatal("got an empty response for cachegroups")
 	}
 	cg = resp[0]
+	if cg.TypeID == nil {
+		t.Error("Cache Group returned by Traffic Ops had null or undefined typeId")
```

Comment:
I think we should change this to fatal here. Its dereferencing `cg.TypeID` on line 551.

Reply:
Yes, that should've been fatal